packages/reactivity/src/ref.ts

```diff
@@ -285,30 +285,22 @@ type UnwrapRefSimple<T> = T extends
 // Extract all known symbols from an object
 // when unwrapping Object the symbols are not `in keyof`, this should cover all the
 // known symbols
-type SymbolExtract<T> = (T extends { [Symbol.asyncIterator]: infer V }
-  ? { [Symbol.asyncIterator]: V }
-  : {}) &
-  (T extends { [Symbol.hasInstance]: infer V }
-    ? { [Symbol.hasInstance]: V }
-    : {}) &
-  (T extends { [Symbol.isConcatSpreadable]: infer V }
-    ? { [Symbol.isConcatSpreadable]: V }
-    : {}) &
-  (T extends { [Symbol.iterator]: infer V } ? { [Symbol.iterator]: V } : {}) &
-  (T extends { [Symbol.match]: infer V } ? { [Symbol.match]: V } : {}) &
-  (T extends { [Symbol.matchAll]: infer V } ? { [Symbol.matchAll]: V } : {}) &
-  (T extends { [Symbol.replace]: infer V } ? { [Symbol.replace]: V } : {}) &
-  (T extends { [Symbol.search]: infer V } ? { [Symbol.search]: V } : {}) &
-  (T extends { [Symbol.species]: infer V } ? { [Symbol.species]: V } : {}) &
-  (T extends { [Symbol.split]: infer V } ? { [Symbol.split]: V } : {}) &
-  (T extends { [Symbol.toPrimitive]: infer V }
-    ? { [Symbol.toPrimitive]: V }
-    : {}) &
-  (T extends { [Symbol.toStringTag]: infer V }
-    ? { [Symbol.toStringTag]: V }
-    : {}) &
-  (T extends { [Symbol.unscopables]: infer V }
-    ? { [Symbol.unscopables]: V }
-    : {})
+type Symbols = {
+  [Symbol.asyncIterator]: unknown
+  [Symbol.hasInstance]: unknown
+  [Symbol.isConcatSpreadable]: unknown
+  [Symbol.iterator]: unknown
+  [Symbol.match]: unknown
+  [Symbol.matchAll]: unknown
+  [Symbol.replace]: unknown
+  [Symbol.search]: unknown
+  [Symbol.species]: unknown
+  [Symbol.split]: unknown
+  [Symbol.toPrimitive]: unknown
+  [Symbol.toStringTag]: unknown
+  [Symbol.unscopables]: unknown
+}
```

Comment:
```suggestion
type Symbols = {
  [K in keyof Symbol]: Symbol[K] extends symbol ? K : never
}[keyof Symbol]
```
This can be further simplified 👀

Reply:
I found that `SymbolExtract` appeared because of: [ref.ts](⁣https://github.com/vuejs/vue-next/blob/master/packages/reactivity/src/ref.ts#L285-L287⁣), but now TypeScript has solved this problem: [TypeScript - 24622⁣](⁣https://github.com/microsoft/TypeScript/issues/24622⁣).
Maybe we don’t need `SymbolExtract`, I’m not sure if my idea is correct.

Reply:
> Maybe we don’t need `SymbolExtract`, I’m not sure if my idea is correct.

I think we need, because the builtin symbols are not unwrapped just the user defined ones, if you update the test
```ts
it('should keep symbols', () => {
    const customSymbol = Symbol()
    const obj = {
      [Symbol.asyncIterator]: { a: 1 },
      [Symbol.unscopables]: { b: '1' },
      [Symbol.match]: ref(1),
      [customSymbol]: { c: [1, 2, 3] }
    }

    const objRef = ref(obj)

    expect(objRef.value[Symbol.asyncIterator]).toBe(obj[Symbol.asyncIterator])
    expect(objRef.value[Symbol.unscopables]).toBe(obj[Symbol.unscopables])
    expect(objRef.value[Symbol.match]).toStrictEqual(ref(1))
    expect(objRef.value[customSymbol]).toStrictEqual(obj[customSymbol])
  })
```

You'll see that it will fail without the `SymbolExtract`

Reply:
Yes, you are right. We do need `SymbolExtract`, but the above test case did not fail locally (no matter the type or execution, there is no error). So I started looking for the reason:

I wrote a small piece of code:

```js
const obj = {
  [Symbol.match]: ref(1),
}
const observed = reactive(obj)
const value = observed[Symbol.match]
```

In the above code, I found that the type of `value` is actually `number & Ref<number>`

![image](https://user-images.githubusercontent.com/11594438/126776758-462fae3f-459e-472d-a45c-0caa6920b318.png)


But I think it should be `Ref<number>`, so maybe our `UnwrappedObject` should be changed to the following:

```ts
type UnwrappedObject<T> = {
  [P in keyof T]: P extends keyof Symbols ? T[P] : UnwrapRef<T[P]>
}
```

If I have any mistakes please point out, thank you very much!

Reply:
That sounds good thank you>

Can you please add some type tests for that? and update the `UnwrappedObject`?

Thank you

Reply:
Sure, I will complete these.